core/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -4090,4 +4139,72 @@ public void testSeqNoGenerator() throws IOException {
         }
     }
 
+    public void testKeepTranslogAfterGlobalCheckpoint() throws Exception {
```

Comment:
Do we need such a complicated test? can we just check that previous commit points are removed we open a new translog?

Reply:
I've simplified this test to verify that we keep only one safe commit.